arch/posix/core/fatal.c

```diff
@@ -13,119 +13,11 @@
 #include <logging/log_ctrl.h>
 #include "posix_soc_if.h"
 
-const NANO_ESF _default_esf = {
-	0xdeadbaad
-};
-
-/**
- *
- * @brief Kernel fatal error handler
- *
- * This routine is called when a fatal error condition is detected
- *
- * The caller is expected to always provide a usable ESF.  In the event that the
- * fatal error does not have a hardware generated ESF, the caller should either
- * create its own or call _Fault instead.
- *
- * @param reason the reason that the handler was called
- * @param pEsf pointer to the exception stack frame
- *
- * @return This function does not return.
- */
-FUNC_NORETURN void z_NanoFatalErrorHandler(unsigned int reason,
-		const NANO_ESF *esf)
+FUNC_NORETURN void z_arch_system_halt(unsigned int reason)
 {
-	LOG_PANIC();
-
-#ifdef CONFIG_PRINTK
-	switch (reason) {
-	case _NANO_ERR_CPU_EXCEPTION:
-	case _NANO_ERR_SPURIOUS_INT:
-		break;
-
-	case _NANO_ERR_INVALID_TASK_EXIT:
-		printk("***** Invalid Exit Software Error! *****\n");
-		break;
-
-
-	case _NANO_ERR_ALLOCATION_FAIL:
-		printk("**** Kernel Allocation Failure! ****\n");
-		break;
-
-	case _NANO_ERR_KERNEL_OOPS:
-		printk("***** Kernel OOPS! *****\n");
-		break;
-
-	case _NANO_ERR_KERNEL_PANIC:
-		printk("***** Kernel Panic! *****\n");
-		break;
-
-#ifdef CONFIG_STACK_SENTINEL
-	case _NANO_ERR_STACK_CHK_FAIL:
-		printk("***** Stack overflow *****\n");
-		break;
-#endif
-	default:
-		printk("**** Unknown Fatal Error %u! ****\n", reason);
-		break;
-	}
-
-#endif
+	ARG_UNUSED(reason);
 
-	void z_SysFatalErrorHandler(unsigned int reason,
-			const NANO_ESF *pEsf);
-	z_SysFatalErrorHandler(reason, esf);
+	posix_print_error_and_exit("fatal error");
```

Comment:
Just a minor thing:
Could you replace this message with something like:
`posix_print_error_and_exit("Exiting due to fatal error\n");`
(This function does not add any extra \n)

Reply:
done.